tests/datadriven/test_reads.py

```diff
@@ -24,7 +59,58 @@ def getDataModelClass(self):
     def getProtocolClass(self):
         return protocol.GAReadGroupSet
 
-    def testFixMe(self):
+    def testGetReadAlignmentsBothRefs(self):
+        # test that querying by both referenceName and referenceId fails
+        with self.assertRaises(exceptions.BadReadsSearchRequestBothRefs):
+            readGroupSet = self._gaObject
+            for readGroup in readGroupSet._readGroups:
+                list(readGroup.getReadAlignments("a", 5, 0, None))
+
+    def testGetReadAlignments(self):
+        # test that searching with no arguments succeeds
+        readGroupSet = self._gaObject
+        for i, readGroup in enumerate(readGroupSet._readGroups):
+            alignments = list(readGroup.getReadAlignments(
+                None, None, None, None))
+            for j, alignment in enumerate(alignments):
+                otherAlignment = self._reads[i][j]
+                self.assertAlignmentsEqual(alignment, otherAlignment)
+
+    def testGetReadAlignmentsRefName(self):
+        # test that searching with a reference name succeeds
         readGroupSet = self._gaObject
-        # TODO
-        self.assertIsNotNone(readGroupSet)
+        for i, readGroup in enumerate(readGroupSet._readGroups):
+            refNames = self._refNames[i]
```

Comment:
A similar point for `self._refNames` depending on the ordering of the readGroups. Maybe a simple local class to gather together all the information we need about a SamFile together into one place would help here?